manifests/ubuntu.pp

```diff
@@ -15,27 +15,31 @@
 class datadog_agent::ubuntu(
   $apt_key = '382E94DE',
   $agent_version = 'latest',
+  $manage_repo = true,
   $other_keys = ['C7A7DA52']
 ) {
 
+  validate_bool($manage_repo)
   ensure_packages(['apt-transport-https'])
   validate_array($other_keys)
 
-  if !$::datadog_agent::skip_apt_key_trusting {
-    $mykeys = concat($other_keys, [$apt_key])
+  if $manage_repo {
+    if !$::datadog_agent::skip_apt_key_trusting {
+      $mykeys = concat($other_keys, [$apt_key])
 
-    ::datadog_agent::ubuntu::install_key { $mykeys:
-      before  => File['/etc/apt/sources.list.d/datadog.list'],
-    }
+      ::datadog_agent::ubuntu::install_key { $mykeys:
+        before  => File['/etc/apt/sources.list.d/datadog.list'],
+      }
 
-  }
+    }
 
-  file { '/etc/apt/sources.list.d/datadog.list':
-    source  => 'puppet:///modules/datadog_agent/datadog.list',
-    owner   => 'root',
-    group   => 'root',
-    notify  => Exec['datadog_apt-get_update'],
-    require => Package['apt-transport-https'],
+    file { '/etc/apt/sources.list.d/datadog.list':
+      source  => 'puppet:///modules/datadog_agent/datadog.list',
+      owner   => 'root',
+      group   => 'root',
+      notify  => Exec['datadog_apt-get_update'],
+      require => Package['apt-transport-https'],
+    }
```

Comment:
Because our package can no longer rely on anything in this logic block you should add: `Package { require => Exec['datadog_apt-get_update']}` so that the update happens before the package is installed